eos/meson-mixing/bq-mixing.cc

```diff
@@ -0,0 +1,121 @@
+ *
+ * This file is part of the EOS project. EOS is free software;
+ * you can redistribute it and/or modify it under the terms of the GNU General
+ * Public License version 2, as published by the Free Software Foundation.
+ *
+ * EOS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#include <eos/meson-mixing/bq-mixing.hh>
+#include <eos/utils/destringify.hh>
+#include <eos/utils/model.hh>
+#include <eos/utils/power_of.hh>
+#include <eos/utils/private_implementation_pattern-impl.hh>
+#include <eos/utils/options-impl.hh>
+
+namespace eos
+{
+    using std::norm;
+
+    template <>
+    struct Implementation<BMixing>
+    {
+        std::shared_ptr<Model> model;
+
+        UsedParameter hbar;
+
+        UsedParameter g_fermi;
+
+        SwitchOption opt_q;
+
+        UsedParameter m_B;
+
+        UsedParameter f_B;
+
+        UsedParameter tau_B;
+
+        UsedParameter R_1;
+        UsedParameter R_2;
+        UsedParameter R_3;
+        UsedParameter R_4;
+        UsedParameter R_5;
+
+        Implementation(const Parameters & p, const Options & o, ParameterUser & u) :
+            model(Model::make(o.get("model", "SM"), p, o)),
+            hbar(p["QM::hbar"], u),
+            g_fermi(p["WET::G_Fermi"], u),
+            opt_q(o, "q", {"s"}),
+            m_B(p["mass::B_" + opt_q.value()], u),
+            f_B(p["decay-constant::B_" + opt_q.value()], u),
+            tau_B(p["life_time::B_" + opt_q.value()], u),
+            R_1(p["B_s<->Bbar_s::R^1"], u),
+            R_2(p["B_s<->Bbar_s::R^2"], u),
+            R_3(p["B_s<->Bbar_s::R^3"], u),
+            R_4(p["B_s<->Bbar_s::R^4"], u),
+            R_5(p["B_s<->Bbar_s::R^5"], u)
+        {
+            u.uses(*model);
+        }
+
+        complex<double> M_12() const
+        {
+            const double mu = 4.2;
+            const auto wc = model->wilson_coefficients_sbsb(mu);
+
+            // cf. [DDHLMSW:2019A]
+            const std::array<complex<double>, 8> contributions{{
+                wc.c1()  * R_1(),
+                wc.c2()  * R_2(),
+                wc.c3()  * R_3(),
+                wc.c4()  * R_4(),
+                wc.c5()  * R_5(),
+                wc.c1p() * R_1(),
+                wc.c2p() * R_2(),
+                wc.c3p() * R_3(),
+            }};
+
+            complex<double> result = 0.0;
+            for (auto & c : contributions)
+            {
+                result += c;
+            }
+
+            // cf. [BBL:1995A], eq. (XVIII.17), p. 153
+            return 4.0 * g_fermi / std::sqrt(2.0) * power_of<2>(model->ckm_tb() * std::conj(model->ckm_ts()))
```

Comment:
Shouldn't we have a g_fermi^2 somewhere? I don't understand your prefactor.

Reply:
The Hamiltonian for F=2 is normalized to G_F^2 * M_W^2, or g'^4 / M_W^2, whatever you prefer. However, one needs to separate the operator from the coefficient and a normalization. The common normalization of the WET in terms of chiral fields is 4 G_F / sqrt(2), which I use here, and which matches the normalization of the Bern basis, except for CKM factors.

The remainder, i.e., G_F * M_W^2, is hidden in the SM value of c1.

Reply:
ahh sorry, it makes sense now